src/Microsoft.AspNetCore.Routing.Abstractions/LinkGenerator.cs

```diff
@@ -6,78 +6,184 @@
 
 namespace Microsoft.AspNetCore.Routing
 {
+    /// <summary>
+    /// Defines a contract to generate URLs to endpoints.
+    /// </summary>
     public abstract class LinkGenerator
     {
+        /// <summary>
+        /// Generates a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <returns>The generated absolute URL.</returns>
         public string GetLink(object values)
         {
             return GetLink(httpContext: null, routeName: null, values, options: null);
         }
 
+        /// <summary>
+        /// Generates a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="options">The <see cref="LinkOptions"/>.</param>
+        /// <returns>The generated absolute URL.</returns>
         public string GetLink(object values, LinkOptions options)
         {
             return GetLink(httpContext: null, routeName: null, values, options);
         }
 
+        /// <summary>
+        /// Tries generating a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="link">The generated absolute URL.</param>
+        /// <returns><c>true</c> if a URL was generated successfully, <c>false</c> otherwise.</returns>
         public bool TryGetLink(object values, out string link)
         {
             return TryGetLink(httpContext: null, routeName: null, values, options: null, out link);
         }
 
+        /// <summary>
+        /// Tries generating a URL with an absolute path.
+        /// </summary>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="options">The <see cref="LinkOptions"/>.</param>
+        /// <param name="link">The generated absolute URL.</param>
+        /// <returns><c>true</c> if a URL was generated successfully, <c>false</c> otherwise.</returns>
         public bool TryGetLink(object values, LinkOptions options, out string link)
         {
             return TryGetLink(httpContext: null, routeName: null, values, options, out link);
         }
 
+        /// <summary>
+        /// Generates a URL with an absolute path.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> associated with current request.</param>
```

Comment:
Does the `HttpContext` have to be the one for the current request?

Reply:
That's something we generally say to make it clear that you're not supposed to just new one up. It's only meaningful when you have a request.